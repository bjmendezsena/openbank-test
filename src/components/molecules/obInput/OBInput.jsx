import React, { useState } from "react";

import { AiOutlineEye, AiOutlineEyeInvisible } from "react-icons/ai";

import "./OBInput.scss";

export const OBInput = ({
  placeHolder = "",
  onChange,
  withIcon = false,
  type = "",
  style = {},
  textLabel = ''
}) => {
  const [visible, setVisible] = useState(false);
  const iconClassName = "OBInput-container-icon";
  const iconSize = 36;

  const inputType = withIcon && visible ? "password" : type;

  return (
    <div className="OBInput">
      <label className="OBInput-label" htmlFor="text">{textLabel}</label>
      <div className="OBInput-container">
        <input
          style={{
            ...style,
          }}
          id="text"
          className="OBInput-container-input"
          placeholder={placeHolder}
          onChange={onChange}
          type={inputType}
          autoComplete="off"
        />

        {withIcon ? (
          visible ? (
            <AiOutlineEye
              className={iconClassName}
              onClick={() => setVisible(false)}
              id="visible-icon"
              size={iconSize}
              color="#002b45"
            />
          ) : (
            <AiOutlineEyeInvisible
              className={iconClassName}
              id="invisible-icon"
              size={iconSize}
              color="#c5d0d3"
              onClick={() => setVisible(true)}
            />
          )
        ) : null}
      </div>
    </div>
  );
};
