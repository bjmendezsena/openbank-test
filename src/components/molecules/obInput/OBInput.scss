@import "../../../styles/base.scss";

.OBInput {
  padding: 0 15px;
  align-items: center;
  &-label {
    font-weight: 400;
    letter-spacing: 1px;
    color: $secondary-color;
  }
  &-container {
    position: relative;
    margin-top: 7px;
    display: flex;
    &-input {
      width: 100%;
      border: none;
      border: 1px solid $opacity;
      border-radius: 5px;
      height: 45px;
      transition: border 0.1s ease-in-out;
      padding: 0 20px;
      font-size: 1rem;
      letter-spacing: 2px;
      color: $secondary-color;

      &::placeholder {
        font-size: 1rem;
        letter-spacing: 2px;
        color: $opacity;
      }
      &:focus {
        border-bottom: solid $orange;
        transition: border 0.1s ease-in-out;
        outline: none;
      }
    }
    &-icon {
      position: relative;
      left: -8vh;
      align-self: center;
      cursor: pointer;
    }
  }
}
