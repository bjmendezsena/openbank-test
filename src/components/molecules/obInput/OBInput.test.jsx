import React from "react";
import { OBInput } from "./OBInput";
import { render, fireEvent } from "@testing-library/react";
import ReactDom from "react-dom";
import { shallow } from "enzyme";


const getWrapper = (props = {}) => {
  return render(<OBInput {...props} />);
};

describe("Test in OPinput component", () => {
  const onChange = jest.fn();
  const placeHolder = "Input";

  it("Should display the document", () => {
    const wrapper = getWrapper({ onChange, placeHolder });
    const { getByPlaceholderText } = wrapper;

    expect(getByPlaceholderText(placeHolder)).toBeInTheDocument();
  });
  it("Should change the value", () => {
    const wrapper = getWrapper({ onChange, placeHolder });
    const newValue = "Hola mundo";
    const { getByPlaceholderText } = wrapper;
    const input = getByPlaceholderText(placeHolder);
    fireEvent.change(input, { target: { value: newValue } });

    expect(input.value).toBe(newValue);
  });

  it("Should display the visible icon", () => {
    const container = document.createElement("div");
    document.body.appendChild(container);

    const withIcon = true;

    const props = { onChange, placeHolder,  withIcon };

    const wrapper = shallow(<OBInput {...props} />);
    const icon = wrapper.find('.OBInput-container-icon').getElement();
    expect(icon).toMatchSnapshot();
  });

  it("Should display the invisible icon", () => {
    const container = document.createElement("div");
    const onTouchVisible = jest.fn();
    document.body.appendChild(container);
    const visible = false;
    const withIcon = true;

    const props = { onChange, placeHolder, visible, withIcon , onTouchVisible};

    ReactDom.render(<OBInput {...props} />, container);
    const icons = container.querySelectorAll("#invisible-icon");

    expect(icons.length).toBe(1);
    document.body.removeChild(container);
    container.remove();
  });
});
