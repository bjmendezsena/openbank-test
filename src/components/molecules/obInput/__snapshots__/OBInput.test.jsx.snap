// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in OPinput component Should display the visible icon 1`] = `
<AiOutlineEyeInvisible
  className="OBInput-container-icon"
  color="#c5d0d3"
  id="invisible-icon"
  onClick={[Function]}
  size={36}
/>
`;
